libs/shared/flagd-js-core/src/lib/feature-flag.ts

```diff
@@ -0,0 +1,202 @@
+import {FlagValue} from '@openfeature/server-sdk';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Flagd flag configuration structure.
+ */
+export class FeatureFlag {
+  private readonly _state: string;
+  private readonly _defaultVariant: string;
+  private readonly _variants: Map<string, FlagValue>;
+  private readonly _targetingString: string;
+
+  constructor(flag: any) {
+    this._state = flag['state'];
+    this._defaultVariant = flag['defaultVariant'];
+    this._variants = new Map<string, FlagValue>(Object.entries(flag['variants']));
+    this._targetingString = JSON.stringify(flag['targeting']);
```

Comment:
Why are you storing this as a string?

Reply:
Targeting handling is not implemented yet and string here is for simplicity